### DISCUSSION TITLE
How to get X and Y coordinates of every node outputted to .csv file

### DISCUSSION THREAD

**Discussion by jackeggemeyer** [idaholab/moose#27419] [Q&A General] [answered]
Hello! I am trying to get moose to output a .cvs file that I need for post processing in MATLAB. I am looking to create a .csv or .txt file that contains the X and Y coordinate of each node and the associated variable called etab0 for each time step. For example, if I had 4 nodes, the output file would be something like:

x   y   eta
1   1   .2
1   2   .4
2   1   .3
2   2    .6

and the next file containing the next data set from the next time step or:

time   x   y   eta
1       1   1   .2
1       1   2   .4
1       2   1   .3
1       2   2    .6
2       1   1   .2
2       1   2   .4
2       2   1   .3
2       2   2    .6

I see the NodalVariableValue Post Processor however it requires a specified node and I want all the nodes as my simulation is more the 4 nodes. Is there a "all" command for this? Thank you

Also: it doesn't need to be the exact format I just need the variable and the associated coordinates for all nodes. And I would also take just the variable associated with each node for each time step and I could write a function in excel/MATLAB to fill in the x and y location.

**Comment by dschwen** [comment]
I could have sworn we already have an object for that, but I cannot find it now. You could write your own derived from `NodalVariableVectorPostprocessor` though.

**Reply by jackeggemeyer** [reply]
Would that involve writing code in C++? I would also be fine with the just the variable being outputted for each node for each time step and I fill in the coordinate values. Is there possibly a post processor for that?

**Reply by maxnezdyur** [reply, marked as answer]
@dschwen I think you are thinking of the https://mooseframework.inl.gov/source/vectorpostprocessors/NodalValueSampler.html?

**Reply by jackeggemeyer** [reply]
@maxnezdyur Thank you so much. That is perfect; I also found a workaround where you upload all the files to ParaView, then use the "save data" button and select CSV, and that also creates a file with x & y cords with associated variables.
